toucan_connectors/toucan_connector.py

```diff
@@ -362,15 +362,15 @@ def get_slice(
             rows from a request, before limit (Snowflake)
         """
         df = self.get_df(data_source, permissions)
-        if limit is not None:
-            return DataSlice(
-                df[offset : offset + limit],
-                stats=DataStats(total_returned_rows=len(df), total_rows=len(df)),
-            )
-        else:
-            return DataSlice(
-                df[offset:], stats=DataStats(total_returned_rows=len(df), total_rows=len(df))
-            )
+        truncated_df = df[offset : offset + limit] if limit is not None else df[offset:]
+        return DataSlice(
+            truncated_df,
```

Comment:
```suggestion
        return DataSlice(
            df[offset : offset + limit] if limit is not None else df[offset:],
```

Reply:
Thx for the proposal, however I find the intermediate variable useful to get what's going on :shrug: